src/v/cloud_topics/reconciler/reconciler_probe.cc

```diff
@@ -61,10 +62,52 @@ void reconciler_probe::setup_metrics() {
           [this] { return _empty_objects_skipped; },
           sm::description(
             "Total objects skipped because no data was available")),
+
+        // Gauges.
         sm::make_gauge(
           "active",
           [this] { return _active; },
           sm::description("Count of active reconciliation rounds")),
+
+        // Histograms.
+        sm::make_histogram(
+          "round_duration_seconds",
+          [this] { return _round_duration.internal_histogram_logform(); },
+          sm::description("Duration of reconciliation rounds")),
```

Comment:
FYI histograms are not a cheap metric. Since we have only one reconciler per core in the system, the increase in volume shouldn't be problematic, but it's something to be careful with.